### DISCUSSION TITLE
Typescript running very slowly (in VSCode)

### DISCUSSION THREAD

**Discussion by sandgraham** [stitchesjs/stitches#623] [Help]
Hi, I've noticed that as I've started building components with stitches, the typescript checking/intellisense in VSCode has started to slow down dramatically (I also use ESLint).

It's pretty common for me to wait a few seconds to see a TS linting error appear or disappear. Autocompletion for variant props or style objects takes a few seconds to appear as well. Often when I open a file using/defining a styled component the fans on my laptop start whirring loudly.

I think it might have something to do with using `StitchesVariants` or the polymorphic typing util- which I have to use often when wrapping a styled component. Importing either generally gets the fans started (but sometimes just using a styled component is enough).

When I started my project with stitches it was pretty awesome how the types seemed to flow through everything. But as the project has grown (still not that big), the sluggish type checking/intellisense issue has become more pronounced.

I'm wondering if others have run into this issue and if so how they might have alleviated it.

**Comment by thiagodebastos** [comment]
@sandgraham I think I've noticed the same thing. I will do some tests to make sure.
On a side note, if you install storybook while using those polymorphic tools, things break - something to do with Storybook changing the ref type to legacyRef...

PS I've never used Github discussions before and it feels weird that this has to be submitted as an 'answer' when I just want to add a comment.

**Comment by peduarte** [comment]
Thanks for that. Good news is, this week we're starting a "TypeScript sprint". Where we'll be addressing many of the known TS related issues, including its performance. 

I'll post an update when this is released. 

Thanks! 🚀

**Reply by sandgraham** [reply]
Ah that's awesome! Very excited for that.

**Reply by anayini** [reply]
@peduarte hope all is well!  Any updates on the Typescript sprint?

**Reply by peduarte** [reply]
It's WIP #659
